CRM/Contribute/Form/ContributionBase.php

```diff
@@ -313,6 +313,17 @@ public function preProcess() {
         throw new CRM_Contribute_Exception_InactiveContributionPageException(ts('The page you requested is currently unavailable.'), $this->_id);
       }
 
+      $endDate = CRM_Utils_Date::processDate(CRM_Utils_Array::value('end_date', $this->_values));
```

Comment:
so I wonder why $this->_values is not in iso format then? -

Reply:
@eileenmcnaughton : it is in iso format as well but needs to be compared to $now 
$endDate = 2021-04-27 00:00:00
$now = 20180523123026

Reply:
@eileenmcnaughton : it is in iso format as well but needs to be compared to $now
```
$this->_values['end_date'] = 2021-04-27 00:00:00
$now = 20180523123026

```

Reply:
OK - so I guess the readability thing is that instead of using standard php functions we have to dig into a custom function to realise it's basically not doing anything a std function doesn't. 

To make it more readable we should make it clear we are comparing like with like - ie preferably compare timestamps

```
if (!empty[ $this->_values['end_date']) {
$timestamp = strtotime(CRM_Utils_Array::value('end_date', $this->_values));
if ( $timestamp < time())
....
```

But if for some reason that is going to be a problem then at least format $this->_values['end_date'] the same way as $now ie.

```
if (!empty[ $this->_values['end_date']) {
$timestamp = strtotime(CRM_Utils_Array::value('end_date', $this->_values));
if ( date('YmdHis', $timestamp)  < date('YmdHis'))
....
```

Reply:
@eileenmcnaughton 
how about using ```CRM_Utils_Date::isoToMysql``` which makes it clearer.

Reply:
Hmm - I feel like for most devs using straight php functions makes is clearer since they are highly used to those & the Civi date functions can be a bit weird & wonderful - but I do agree that is a lot clearer than 'processDate'

Is it really ISO format ? I mean there would never  timezone info on the date?